integrations/standalone/tests/pageobjects/CodeEditor.ts

```diff
@@ -0,0 +1,50 @@
+import { Locator, Page, expect } from '@playwright/test';
+import { FocusCodeEditorUtil } from '../utils/code-editor-util';
+
+export class CodeEditor {
+  private readonly label: string | undefined;
+  private readonly parentLocator: Locator;
+  private readonly page: Page;
+  private readonly contentAssist: Locator;
+  private readonly value: Locator;
+
+  constructor(page: Page, parentLocator: Locator, activateOnFocus: boolean, label: string | undefined) {
+    this.page = page;
+    this.parentLocator = parentLocator;
+    this.label = label;

```

Comment:
Instead of this, you can also do it like this:
```js
export class CodeEditor {
  private readonly contentAssist: Locator;
  private readonly value: Locator;
  constructor(readonly page: Page, readonly parentLocator: Locator, activateOnFocus: boolean, readonly label?: string) {
    this.contentAssist = parentLocator.locator('div.suggest-widget');
    this.value = activateOnFocus ? this.locator() : this.locator().getByRole('textbox');
  }
```

If you use the readonly param in the constructor it automatically creates a parameter for you and does the assignment.